### PR TITLE
Add funders, projects, and geography units to auth service schema

### LINKED ISSUE #12: Add funders, projects, and geography units to auth-service schema
## Summary
- Funders & Projects master-data tables (ERD Appendix A.1) with tests.
- Self-referencing `geography_units` table scoped to the SRS's 7-level hierarchy (State -> District -> Block -> PHC -> Sub-centre -> Village -> Pada), with tests.
- `form_definitions`/`form_versions` migration applied to visit-form-service's schema, with 8 tests.

## Branch
`pp/mvp`

## Resolved by
#7 (open)

Status: in progress.

### PR DESCRIPTION
**Pull request by Pemmaiah544**
**Funders & Projects (auth-service)**: new master-data tables, matching ERD Appendix A.1 exactly, with tests.
**Geography hierarchy (auth-service)**: self-referencing geography_units table, scoped to the SRS's 7-level hierarchy (State → District → Block → PHC → Sub-centre → Village → Pada), with tests.
**Added the form_definitions/form_versions migration (this sprint's actual deliverable)** — applied to visit-form-service's own schema, with 8 tests covering: creation, unique form_code, soft-delete, valid/invalid form_definition_id references, unique (form_definition_id, version_no), and required checksum.
**Fixed a gap surfaced by the merge:** GeographyUnit and its enums were missing the @@schema annotation the new multiSchema mode requires — added, matching the rest of the file.
**Fixed a second gap**: the geography_units migration had been committed but never actually applied to any database — applied it to the correct schema.
**Fixed a missing dependency:** jose (JWT signing library) was used in code merged from another branch but never added to package.json — added and installed.

Closes #12.

Related to #84

### CODE REVIEW

File: apps/auth-service/src/auth/dto/login.dto.ts [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+/**
+ * Login is username + password for every role, per SRS FR-S-1.1 ("Sakhi must
+ * authenticate using Username and password"), applied to all roles rather
+ * than SAKHI only. mobileNumber remains a real `users` column (per the ERD)
+ * but is no longer a login credential.
+ */
 export const loginSchema = z
   .object({
-    mobileNumber: mobileNumberSchema,
+    username: z.string().trim().min(1).max(50),
```
Comment (palani-ND) [inline apps/auth-service/src/auth/dto/login.dto.ts]: This makes the public API username-only, but AuthRepository.findUserByIdentifier() still supports both username and mobile number and its comment says login accepts either. Could we align those layers one way or the other? If username-only is the intended requirement, please rename the repository method/comment and add a test proving mobile-number login is rejected at the API boundary.

File: apps/auth-service/src/auth/auth.repository.ts [hunk 2]
```diff
@@ -4,9 +4,10 @@ import type { PrismaService } from '../prisma/prisma.service';
 export class AuthRepository {
   constructor(private readonly prisma: PrismaService) {}
 
-  findUserByMobileNumber(mobileNumber: string) {
-    return this.prisma.user.findUnique({
-      where: { mobileNumber },
+  /** Looks up a user by mobile number or username — login accepts either. */
+  findUserByIdentifier(identifier: string) {
+    return this.prisma.user.findFirst({
```
Comment (palani-ND) [inline apps/auth-service/src/auth/auth.repository.ts]: Both mobileNumber and username are unique, so findFirst weakens the query contract a bit here. If login is username-only, findUnique({ where: { username: identifier } }) would better match the schema. If login supports either identifier, consider branching and using findUnique on each path so the lookup semantics stay aligned with the uniqueness constraints.

File: apps/auth-service/prisma/migrations/20260714124235_add_geography_units/migration.sql [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    "deleted_at" TIMESTAMP(3),
+
+    CONSTRAINT "geography_units_pkey" PRIMARY KEY ("geography_unit_id")
+);
+
+-- CreateIndex
+CREATE UNIQUE INDEX "geography_units_parent_id_geo_type_geo_code_key" ON "geography_units"("parent_id", "geo_type", "geo_code");
+
+-- AddForeignKey
+ALTER TABLE "geography_units" ADD CONSTRAINT "geography_units_parent_id_fkey" FOREIGN KEY ("parent_id") REFERENCES "geography_units"("geography_unit_id") ON DELETE SET NULL ON UPDATE CASCADE;
```
Comment (palani-ND) [inline apps/auth-service/prisma/migrations/20260714124235_add_geography_units/migration.sql:29]: This sets up the self-reference, but it doesn’t enforce the claimed 7-level hierarchy. As written, the DB would still allow invalid parent/child combinations like PADA -> STATE or DISTRICT under PHC. If the SRS hierarchy is mandatory, should we enforce valid transitions in the schema (trigger/check) or explicitly defer that to application validation?

File: apps/auth-service/src/prisma/geography.schema.spec.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        data: { geoType: 'STATE', name: `Maharashtra-${Date.now()}` },
+      }),
+    );
+
+    expect(state.parentId).toBeNull();
+    expect(state.status).toBe('ACTIVE');
+    expect(state.isDeleted).toBe(false);
+  });
+
+  it('builds a full 7-level chain, each linked to its parent', async () => {
```
Comment (palani-ND) [inline apps/auth-service/src/prisma/geography.schema.spec.ts:49]: Nice happy-path coverage. I think we still need a negative test for invalid hierarchy relationships once the constraint exists—for example proving that illegal parent/child combinations are rejected, not just that a valid chain can be inserted.

File: apps/auth-service/src/prisma/geography.schema.spec.ts [hunk 5]
```diff
@@ -0,0 +1,184 @@
+import { PrismaService } from './prisma.service';
+
+/**
+ * Verifies the geography_units migration itself (self-referencing hierarchy,
+ * composite unique constraint, FK behaviour) — there is no API layer for this
+ * domain yet, so these tests exercise the Prisma client directly against the
+ * real schema. Scoped to the SRS's 7-level hierarchy only (State > District >
+ * Block > PHC > Sub-centre > Village > Pada) — no Taluka, no Panchayat.
+ */
+describe('geography schema (geography_units)', () => {
```
Comment (palani-ND) [inline apps/auth-service/src/prisma/geography.schema.spec.ts]: This suite appears to hit a real database unconditionally, while master-data.schema.spec.ts is guarded behind process.env.DATABASE_URL ? describe : describe.skip. Could we standardize that pattern across the DB-backed schema tests so CI/local execution is predictable?

File: apps/visit-form-service/src/prisma/formDefinition.schema.spec.ts [hunk 6]
```diff
@@ -0,0 +1,247 @@
+import { PrismaService } from './prisma.service';
+
+/**
+ * Verifies the form_definitions / form_versions migration itself (unique
+ * form_code, versioning constraint, FK behaviour) — there is no API layer for
+ * this domain yet, so these tests exercise the Prisma client directly against
+ * the real schema.
+ */
+describe('form schema (form_definitions, form_versions)', () => {
```
Comment (palani-ND) [inline apps/visit-form-service/src/prisma/formDefinition.schema.spec.ts]: This looks like another real-DB schema test suite, but I don’t see the same environment guard used in master-data.schema.spec.ts. Could we make the execution contract consistent across these integration-style schema tests so they don’t behave differently per environment?

File: apps/visit-form-service/src/prisma/formDefinition.schema.spec.ts [hunk 7]
```diff
@@ -0,0 +1,254 @@
+import { PrismaService } from './prisma.service';
+
+/**
+ * Verifies the form_definitions / form_versions migration itself (unique
+ * form_code, versioning constraint, FK behaviour) — there is no API layer for
+ * this domain yet, so these tests exercise the Prisma client directly against
+ * the real schema.
+ *
+ * Requires a real, migrated database (DATABASE_URL/DIRECT_URL), which CI does
```
Comment (palani-ND) [inline apps/visit-form-service/src/prisma/formDefinition.schema.spec.ts:9]: This looks like another real-DB schema test suite, but I don’t see the same environment guard used in master-data.schema.spec.ts. Could we make the execution contract consistent across these integration-style schema tests so they don’t behave differently per environment?

Comment (Pemmaiah544) [reply apps/visit-form-service/src/prisma/formDefinition.schema.spec.ts:9]: Already handled — the DATABASE_URL guard (const describeIfDatabaseConfigured = process.env.DATABASE_URL ? describe : describe.skip;) is already present on line 14 of this file, added in commit 3904c54 which is part of this PR

File: apps/auth-service/prisma/migrations/20260714124235_add_geography_units/migration.sql (hunk 3 above)
[the remark this answers is not in the document]
Comment (Pemmaiah544) [reply apps/auth-service/prisma/migrations/20260714124235_add_geography_units/migration.sql:29]: fixed

File: apps/approval-service/.env.example [hunk 8]
```diff
@@ -1,6 +1,6 @@
 NODE_ENV=development
 PORT=3007
 LOG_LEVEL=debug
-DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:5432/postgres?schema=approval-service
+DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:6543/postgres?schema=approval-service&pgbouncer=true&connection_limit=1
```
[the remark this answers is not in the document]
Comment (Pemmaiah544) [reply apps/approval-service/.env.example]: fixed

File: apps/beneficiary-service/.env.example [hunk 9]
```diff
@@ -1,6 +1,6 @@
 NODE_ENV=development
 PORT=3001
 LOG_LEVEL=debug
-DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:5432/postgres?schema=beneficiary
+DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:6543/postgres?schema=beneficiary&pgbouncer=true&connection_limit=1
```
[the remark this answers is not in the document]
Comment (Pemmaiah544) [reply apps/beneficiary-service/.env.example]: fixed

File: apps/closure-reopen-service/.env.example [hunk 10]
```diff
@@ -1,6 +1,6 @@
 NODE_ENV=development
 PORT=3006
 LOG_LEVEL=debug
-DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:5432/postgres?schema=closure-reopen-service
+DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:6543/postgres?schema=closure-reopen-service&pgbouncer=true&connection_limit=1
```
[the remark this answers is not in the document]
Comment (Pemmaiah544) [reply apps/closure-reopen-service/.env.example]: fixed

File: apps/incentive-wages-service/.env.example [hunk 11]
```diff
@@ -1,6 +1,6 @@
 NODE_ENV=development
 PORT=3008
 LOG_LEVEL=debug
-DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:5432/postgres?schema=incentive-wages-service
+DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:6543/postgres?schema=incentive-wages-service&pgbouncer=true&connection_limit=1
```
[the remark this answers is not in the document]
Comment (Pemmaiah544) [reply apps/incentive-wages-service/.env.example]: fixed

File: apps/media-service/.env.example [hunk 12]
```diff
@@ -1,6 +1,6 @@
 NODE_ENV=development
 PORT=3011
 LOG_LEVEL=debug
-DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:5432/postgres?schema=media-service
+DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:6543/postgres?schema=media-service&pgbouncer=true&connection_limit=1
```
[the remark this answers is not in the document]
Comment (Pemmaiah544) [reply apps/media-service/.env.example]: fixed

File: apps/notification-escalation-service/.env.example [hunk 13]
```diff
@@ -1,6 +1,6 @@
 NODE_ENV=development
 PORT=3009
 LOG_LEVEL=debug
-DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:5432/postgres?schema=notification-escalation-service
+DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:6543/postgres?schema=notification-escalation-service&pgbouncer=true&connection_limit=1
```
[the remark this answers is not in the document]
Comment (Pemmaiah544) [reply apps/notification-escalation-service/.env.example]: fixed

File: apps/risk-referral-service/.env.example [hunk 14]
```diff
@@ -1,6 +1,6 @@
 NODE_ENV=development
 PORT=3005
 LOG_LEVEL=debug
-DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:5432/postgres?schema=risk-referral-service
+DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:6543/postgres?schema=risk-referral-service&pgbouncer=true&connection_limit=1
```
[the remark this answers is not in the document]
Comment (Pemmaiah544) [reply apps/risk-referral-service/.env.example]: fixed

File: apps/rules-service/.env.example [hunk 15]
```diff
@@ -1,6 +1,6 @@
 NODE_ENV=development
 PORT=3004
 LOG_LEVEL=debug
-DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:5432/postgres?schema=rules-service
+DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:6543/postgres?schema=rules-service&pgbouncer=true&connection_limit=1
```
[the remark this answers is not in the document]
Comment (Pemmaiah544) [reply apps/rules-service/.env.example]: fixed

File: apps/visit-form-service/.env.example [hunk 16]
```diff
@@ -1,6 +1,6 @@
 NODE_ENV=development
 PORT=3003
 LOG_LEVEL=debug
-DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:5432/postgres?schema=visit-form-service
+DATABASE_URL=postgresql://postgres.PROJECT_REF:REPLACE_ME@aws-0-ap-northeast-1.pooler.supabase.com:6543/postgres?schema=visit-form-service&pgbouncer=true&connection_limit=1
```
[the remark this answers is not in the document]
Comment (Pemmaiah544) [reply apps/visit-form-service/.env.example]: fixed
